src/sentry/notifications/notifications/base.py

```diff
@@ -227,13 +232,18 @@ def get_notification_providers(self) -> Iterable[ExternalProviders]:
     def filter_to_accepting_recipients(
         self, recipients: Iterable[RpcActor]
     ) -> Mapping[ExternalProviders, Iterable[RpcActor]]:
+        assert self.notification_setting_type is not None
```

Comment:
cleaned up type errors related to notification settings 2.0

Reply:
Will this exception be handled by the caller?
Should this be an if condition with an early return?
Do we have a test case that handles this?
At first glance, this change makes me a bit iffy.